mm2src/coins/utxo.rs

```diff
@@ -141,6 +142,9 @@ pub type HistoryUtxoTxMap = HashMap<H256Json, HistoryUtxoTx>;
 pub type MatureUnspentMap = HashMap<Address, MatureUnspentList>;
 pub type RecentlySpentOutPointsGuard<'a> = AsyncMutexGuard<'a, RecentlySpentOutPoints>;
 pub type UtxoHDAddress = HDAddress<Address, Public>;
+pub type ScripthashNotificationSender = Option<Arc<AsyncMutex<AsyncSender<String>>>>;
```

Comment:
The Sender is `Clone`, so Arc and Mutex might be not required, please consider the following diff:
```diff
diff --git a/mm2src/coins/utxo.rs b/mm2src/coins/utxo.rs
index 1d203be26..fb904c2be 100644
--- a/mm2src/coins/utxo.rs
+++ b/mm2src/coins/utxo.rs
@@ -142,7 +142,7 @@ pub type HistoryUtxoTxMap = HashMap<H256Json, HistoryUtxoTx>;
 pub type MatureUnspentMap = HashMap<Address, MatureUnspentList>;
 pub type RecentlySpentOutPointsGuard<'a> = AsyncMutexGuard<'a, RecentlySpentOutPoints>;
 pub type UtxoHDAddress = HDAddress<Address, Public>;
-pub type ScripthashNotificationSender = Option<Arc<AsyncMutex<AsyncSender<String>>>>;
+pub type ScripthashNotificationSender = Option<AsyncSender<String>>;
 
 type ScripthashNotificationReceiver = Option<Arc<AsyncMutex<AsyncReceiver<String>>>>;
 
diff --git a/mm2src/coins/utxo/rpc_clients.rs b/mm2src/coins/utxo/rpc_clients.rs
index e4af251c4..052ff5cae 100644
--- a/mm2src/coins/utxo/rpc_clients.rs
+++ b/mm2src/coins/utxo/rpc_clients.rs
@@ -2489,7 +2489,7 @@ fn rx_to_stream(rx: mpsc::Receiver<Vec<u8>>) -> impl Stream<Item = Vec<u8>, Erro
 async fn electrum_process_json(
     raw_json: Json,
     arc: &JsonRpcPendingRequestsShared,
-    scripthash_notification_sender: &ScripthashNotificationSender,
+    scripthash_notification_sender: ScripthashNotificationSender,
 ) {
     // detect if we got standard JSONRPC response or subscription response as JSONRPC request
     #[derive(Deserialize)]
@@ -2530,9 +2530,9 @@ async fn electrum_process_json(
                         },
                     };
 
-                    if let Some(sender) = scripthash_notification_sender {
+                    if let Some(mut sender) = scripthash_notification_sender {
                         debug!("Sending scripthash notification");
-                        if sender.lock().await.try_send(scripthash.to_string()).is_err() {
+                        if sender.try_send(scripthash.to_string()).is_err() {
                             error!("Failed sending scripthash notification");
                             return;
                         };
@@ -2564,7 +2564,7 @@ async fn electrum_process_json(
 async fn electrum_process_chunk(
     chunk: &[u8],
     arc: &JsonRpcPendingRequestsShared,
-    scripthash_notification_sender: &ScripthashNotificationSender,
+    scripthash_notification_sender: ScripthashNotificationSender,
 ) {
     // we should split the received chunk because we can get several responses in 1 chunk.
     let split = chunk.split(|item| *item == b'\n');
@@ -2578,7 +2578,7 @@ async fn electrum_process_chunk(
                     return;
                 },
             };
-            electrum_process_json(raw_json, arc, scripthash_notification_sender).await
+            electrum_process_json(raw_json, arc, scripthash_notification_sender.clone()).await
         }
     }
 }
@@ -2785,7 +2785,7 @@ async fn connect_loop<Spawner: SpawnFuture>(
                     event_handlers.on_incoming_response(buffer.as_bytes());
                     last_chunk.store(now_ms(), AtomicOrdering::Relaxed);
 
-                    electrum_process_chunk(buffer.as_bytes(), &responses, &scripthash_notification_sender).await;
+                    electrum_process_chunk(buffer.as_bytes(), &responses, scripthash_notification_sender.clone()).await;
                     buffer.clear();
                 }
             }
diff --git a/mm2src/mm2_core/src/mm_ctx.rs b/mm2src/mm2_core/src/mm_ctx.rs
index d7569678e..a1acb3c33 100644
--- a/mm2src/mm2_core/src/mm_ctx.rs
+++ b/mm2src/mm2_core/src/mm_ctx.rs
@@ -132,10 +132,7 @@ pub struct MmCtx {
     pub nft_ctx: Mutex<Option<Arc<dyn Any + 'static + Send + Sync>>>,
 }
 
-type ScripthashNotificationHandlers = Option<(
-    Arc<AsyncMutex<AsyncSender<String>>>,
-    Arc<AsyncMutex<AsyncReceiver<String>>>,
-)>;
+type ScripthashNotificationHandlers = Option<(AsyncSender<String>, Arc<AsyncMutex<AsyncReceiver<String>>>)>;
 
 impl MmCtx {
     pub fn with_log_state(log: LogState) -> MmCtx {
@@ -579,7 +576,7 @@ impl MmArc {
     pub fn get_scripthash_notification_handlers(&self) -> ScripthashNotificationHandlers {
         if self.event_stream_configuration.is_some() {
             let (sender, receiver): (AsyncSender<String>, AsyncReceiver<String>) = futures::channel::mpsc::channel(1);
-            Some((Arc::new(AsyncMutex::new(sender)), Arc::new(AsyncMutex::new(receiver))))
+            Some((sender, Arc::new(AsyncMutex::new(receiver))))
         } else {
             None
         }
```